platform/ui/src/components/content/viewportDownloadForm/ViewportDownloadForm.js

```diff
@@ -129,6 +142,46 @@ const ViewportDownloadForm = ({
     maximumSize,
   ]);
 
+  useEffect(() => {
+    if (refreshViewport.current !== null) {
+      clearTimeout(refreshViewport.current);
+    }
+
+    refreshViewport.current = setTimeout(() => {
+      refreshViewport.current = null;
+      loadAndUpdateViewports();
+    }, REFRESH_VIEWPORT_TIMEOUT);
```

Comment:
This is effectively a debounce. You could use our `useDebounce` hook? It lives in `ui/src/hooks`

Reply:
The useDebounce on our hooks is limited to update the provided state with a timeout, we cannot pass a function to be debaunced.